beacon_node/beacon_chain/src/light_client_optimistic_update_verification.rs

```diff
@@ -67,38 +65,15 @@ impl<T: BeaconChainTypes> VerifiedLightClientOptimisticUpdate<T> {
     /// Returns `Ok(Self)` if the `light_client_optimistic_update` is valid to be (re)published on the gossip
     /// network.
     pub fn verify(
-        light_client_optimistic_update: LightClientOptimisticUpdate<T::EthSpec>,
+        rcv_optimistic_update: LightClientOptimisticUpdate<T::EthSpec>,
         chain: &BeaconChain<T>,
         seen_timestamp: Duration,
     ) -> Result<Self, Error> {
-        let gossiped_optimistic_slot = light_client_optimistic_update.attested_header.beacon.slot;
-        let one_third_slot_duration = Duration::new(chain.spec.seconds_per_slot / 3, 0);
-        let signature_slot = light_client_optimistic_update.signature_slot;
-        let start_time = chain.slot_clock.start_of(signature_slot);
-        let mut latest_seen_optimistic_update = chain.latest_seen_optimistic_update.lock();
-
-        let head = chain.canonical_head.cached_head();
-        let head_block = &head.snapshot.beacon_block;
-        let attested_block_root = head_block.message().parent_root();
-        let attested_block = chain
-            .get_blinded_block(&attested_block_root)?
-            .ok_or(Error::FailedConstructingUpdate)?;
-
-        let attested_state = chain
-            .get_state(&attested_block.state_root(), Some(attested_block.slot()))?
-            .ok_or(Error::FailedConstructingUpdate)?;
-        let latest_seen_optimistic_update_slot = match latest_seen_optimistic_update.as_ref() {
-            Some(update) => update.attested_header.beacon.slot,
-            None => Slot::new(0),
-        };
-
-        // verify that no other optimistic_update with a lower or equal
-        // optimistic_header.slot was already forwarded on the network
-        if gossiped_optimistic_slot <= latest_seen_optimistic_update_slot {
-            return Err(Error::OptimisticUpdateAlreadySeen);
-        }
```

Comment:
Why is this check no longer required?

Reply:
This check is redundant with the equality done below

https://github.com/sigp/lighthouse/blob/90cf69e98fe0ec1765418aad3b4a09c8dd8d744a/beacon_node/beacon_chain/src/light_client_optimistic_update_verification.rs#L94-L96

Reply:
Nice, we can probably get rid of the `Error` enum variants too
https://github.com/sigp/lighthouse/blob/a5ba62da2caa789941ed4354c35449eb7fef2efb/beacon_node/beacon_chain/src/light_client_finality_update_verification.rs#L17-L18